x-pack/plugin/watcher/src/main/java/org/elasticsearch/xpack/watcher/WatcherIndexingListener.java

```diff
@@ -119,8 +124,8 @@ public void postIndex(ShardId shardId, Engine.Index operation, Engine.IndexResul
                 }
 
                 boolean shouldBeTriggered = shardAllocationConfiguration.shouldBeTriggered(watch.id());
-                if (shouldBeTriggered) {
-                    if (watch.status().state().isActive()) {
+                if (shouldBeTriggered && EnumSet.of(WatcherState.STOPPING, WatcherState.STOPPED).contains(watcherState.get()) == false) {
```

Comment:
I manually tested that when the service is stopped, and you add a new watch, it does indeed get added (as expected) upon startup.